ygot/render.go

```diff
@@ -282,6 +452,39 @@ func leavesToNotifications(leaves map[*path]interface{}, ts int64, prefix []inte
 	return []*gnmipb.Notification{n}, nil
 }
 
+// stripPrefix removes the prefix pfx from the supplied path, and returns the more
+// specific path elements of the path. If the prefix is invalid for the path an
+// error is return.
+func stripPrefix(path *pathWrapper, pfx *pathWrapper) (*pathWrapper, error) {
+	if path.pathFormat != pfx.pathFormat {
+		return nil, fmt.Errorf("mismatched path formats in prefix and path, %v != %s", path.pathFormat, pfx.pathFormat)
+	}
+
+	switch path.pathFormat {
```

Comment:
Optional: a nice pattern to follow here is to have stripPrefix() top level, then
if path.IsStringSlicePath() {
return stripPrefixStringSlicePath()
}
return stripPrefixPathElementPath()

Reply:
This ended up being relatively low complexity, but I agree this pattern works.